src/WiFiManager.cpp

```diff
@@ -1,228 +1,206 @@
 #include "WiFiManager.h"
 
 #include "CaptivePortal.h"
+#include "Mappers/EspWiFiTypesMapper.h"
 #include "VisualStateManager.h"
+#include "WiFiCredentials.h"
+#include "WiFiScanManager.h"
 
 #include <ArduinoJson.h>
 #include <LittleFS.h>
 #include <WiFi.h>
 
+#include <esp_wifi_types.h>
+
 #include <vector>
 
+#include <nonstd/span.hpp>
+
 const char* const TAG = "WiFiManager";
 
 using namespace OpenShock;
 
-struct WifiCredentials {
-  String ssid;
-  String password;
-  std::uint16_t wifiIndex;
-  std::uint8_t attempts;
-};
-static std::vector<WifiCredentials> s_wifiCredentials;
-static WiFiState s_wifiState;
-
-void SetWiFiState(WiFiState state) {
-  s_wifiState = state;
-  VisualStateManager::SetWiFiState(state);
+void _broadcastWifiAddNetworkSuccess(const char* ssid) {
+  DynamicJsonDocument doc(64);
+  doc["type"]    = "wifi";
+  doc["subject"] = "add_network";
+  doc["status"]  = "success";
+  doc["ssid"]    = ssid;
+  CaptivePortal::BroadcastMessageJSON(doc);
+}
+void _broadcastWifiAddNetworkError(const char* error) {
+  DynamicJsonDocument doc(64);
+  doc["type"]    = "wifi";
+  doc["subject"] = "add_network";
+  doc["status"]  = "error";
+  doc["error"]   = error;
+  CaptivePortal::BroadcastMessageJSON(doc);
 }
 
-bool SaveCredentials() {
-  File file = LittleFS.open("/networks", FILE_WRITE);
-  if (!file) {
-    ESP_LOGE(TAG, "Failed to open networks file for writing");
-    return false;
-  }
+struct WiFiNetwork {
+  char ssid[33];
+  std::uint8_t bssid[6];
+  std::uint8_t channel;
+  std::int8_t rssi;
+  wifi_auth_mode_t authmode;
```

Comment:
`authMode`?

Reply:
authenticationmode for the wifi network.

'Open' | 'WEP' | 'WPA PSK' | 'WPA2 PSK' | 'WPA/WPA2 PSK' | 'WPA2 Enterprise' | 'WPA3 PSK' | 'WPA2/WPA3 PSK' | 'WAPI PSK'

Reply:
I meant `authmode` => `authMode`

Reply:
Changed.

Reply:
Changed.